code/game/objects/items/candle.dm

```diff
@@ -79,3 +79,50 @@
 		var/obj/item/projectile/beam/P = Proj
 		if(P.damage != 0)
 			light("", 1)
+
+
+/obj/item/candle/holo
+	name = "holo candle"
+	desc = "A small disk projecting the image of a candle, used for futuristic lighting. It has a multitool port on it for changing colors."
+	icon_state = "holocandle_red"
+	//item_state = "candle1"
+	heat_production = 0
+	source_temperature = 0
+	light_color = LIGHT_COLOR_FIRE
+	wax = "red" //Repurposed var for the "wax" color.
+
+/obj/item/candle/holo/update_icon()
+	switch(wax)
+		if("red")
+			light_color = LIGHT_COLOR_FIRE
+		if("blue")
+			light_color = LIGHT_COLOR_BLUE
+		if("purple")
+			light_color = LIGHT_COLOR_PURPLE
+		if("green")
+			light_color = LIGHT_COLOR_GREEN
+		if("yellow")
+			light_color = LIGHT_COLOR_YELLOW
+	icon_state = "holocandle_[wax][lit ? "_lit" : ""]"
+
+/obj/item/candle/holo/attack_self(mob/user)
+	lit = !lit
+	update_icon()
+	light("<span class='notice'>[user] flips \the [src]'s switch.</span>")
+
+/obj/item/candle/attackby(obj/item/weapon/W, mob/user)
```

Comment:
SNIPSSSSSSS
you're overwriting `candle/attackby` instead of doing `candle/holo/attackby`
it runtimes when used on a normal candle